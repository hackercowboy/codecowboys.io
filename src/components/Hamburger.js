import React from 'react';
import PropTypes from 'prop-types';

function Hamburger({ onClick, open }) {
  return (
    <button className="hamburger" type="button" onClick={onClick}>
      <span className="hamburger-box">
        <span className={open ? 'hamburger-inner open' : 'hamburger-inner'} />
      </span>
    </button>
  );
}

Hamburger.propTypes = {
  onClick: PropTypes.func.isRequired,
  open: PropTypes.bool,
};

Hamburger.defaultProps = {
  open: false,
};

export default Hamburger;
