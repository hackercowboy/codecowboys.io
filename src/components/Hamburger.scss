@import '../styles/colors';

.hamburger {
  background-color: transparent;
  border: 0;
  color: inherit;
  cursor: pointer;
  display: inline-block;
  font: inherit;
  margin: 0;
  overflow: visible;

  text-transform: none;
  transition-duration: 0.15s;

  transition-property: opacity, filter;
  transition-timing-function: linear;

}

.hamburger-box {
  display: inline-block;
  height: 4px * 3 + 6px * 2;
  position: relative;
  width: 40px;
}

.hamburger-inner {
  display: block;
  top: 50%;

  transition-duration: 0.22s;
  transition-timing-function: cubic-bezier(0.55, 0.055, 0.675, 0.19);

  &,
  &::before,
  &::after {
    background-color: $text-color;
    border-radius: 4px;
    height: 4px;
    position: absolute;
    transition-duration: 0.15s;
    transition-property: transform;
    transition-timing-function: ease;
    width: 40px;
  }

  &::before,
  &::after {
    content: '';
    display: block;
  }

  &::before {
    top: -10px;
  }

  &::after {
    bottom: -10px;
  }

  &::before {
    transition: top 0.1s 0.25s ease-in,
      opacity 0.1s ease-in;
  }

  &::after {
    transition: bottom 0.1s 0.25s ease-in,
      transform 0.22s cubic-bezier(0.55, 0.055, 0.675, 0.19);
  }
}

.hamburger-inner.open {
  transform: rotate(225deg);
  transition-delay: 0.12s;
  transition-timing-function: cubic-bezier(0.215, 0.61, 0.355, 1);

  &::before {
    opacity: 0;
    top: 0;
    transition: top 0.1s ease-out,
      opacity 0.1s 0.12s ease-out;
  }

  &::after {
    bottom: 0;
    transform: rotate(-90deg);
    transition: bottom 0.1s ease-out,
      transform 0.22s 0.12s cubic-bezier(0.215, 0.61, 0.355, 1);
  }
}

@media (min-width : 576px) {
  .hamburger {
    display: none;
  }
}

