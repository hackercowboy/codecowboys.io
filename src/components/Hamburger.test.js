import React from 'react';
import { shallow } from 'enzyme';

import Hamburger from './Hamburger';

describe('<Hamburger/>', () => {
  it('should render', () => {
    const wrapper = shallow(<Hamburger />);
    expect(wrapper.debug()).toMatchSnapshot();
  });
});
