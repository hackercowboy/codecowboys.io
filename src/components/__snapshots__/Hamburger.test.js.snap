// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Hamburger/> should render 1`] = `
"<button className=\\"hamburger\\" type=\\"button\\" onClick={[undefined]}>
  <span className=\\"hamburger-box\\">
    <span className=\\"hamburger-inner\\" />
  </span>
</button>"
`;
